File: bubble_sort/bubble_sort.go
// Bubble Sort in Go
//
package main

import (
    "fmt"
    "math/rand"
    "time"
)

// Order an array using the Bubble Sort method
func bubble_sort(array []int) {
    var j int
    var swaped bool
    for i := range array {
        swaped = false
        for j=0; j<len(array) - i - 1; j++ {
            if array[j] > array[j+1] {
                array[j+1], array[j] = array[j], array[j+1]
                swaped = true
            }
        }
        if !swaped { break }
    }
}

// The main function
func main() {
    rand.Seed(time.Now().UnixNano())
    arr := make([]int, 10);
    for i:=0; i<len(arr); i++ {
        arr[i] = rand.Intn(99)
    }
    fmt.Println(arr)
    bubble_sort(arr)
    fmt.Println(arr)
}

File: bubble_sort/bubble_sort.py
""" Bubble Sort in Python
"""
from random import randint

def bubble_sort(array:list):
    """ Sort array using Bubble Sort technique """
    size = len(array)
    for i in range(size):
        swaped = False
        for j in range(size - i - 1):
            if array[j+1] < array[j]:
                array[j], array[j+1] = array[j+1], array[j]
                swaped = True
        if not swaped:
            break

if __name__ == "__main__":
    arr = [randint(0, 99) for _ in range(10)]
    print(arr)
    bubble_sort(arr)
    print(arr)
